patches/server/1005-PreSpawnerSearchEvent.patch

```diff
@@ -0,0 +1,61 @@
+From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
+From: TreemanKing <67459602+TreemanKing@users.noreply.github.com>
+Date: Wed, 16 Aug 2023 11:53:44 +1000
+Subject: [PATCH] PreSpawnerSearchEvent
+
+
+diff --git a/src/main/java/net/minecraft/world/level/BaseSpawner.java b/src/main/java/net/minecraft/world/level/BaseSpawner.java
+index 633500aefd515df5dadda3802b94079f75a03fa0..6d22659cf13b6bef32f0d058a6489cf1d365d642 100644
+--- a/src/main/java/net/minecraft/world/level/BaseSpawner.java
++++ b/src/main/java/net/minecraft/world/level/BaseSpawner.java
+@@ -1,6 +1,8 @@
+ package net.minecraft.world.level;
+ 
+ import com.mojang.logging.LogUtils;
++
++import java.util.Collection;
+ import java.util.Optional;
+ import java.util.function.Function;
+ import javax.annotation.Nullable;
+@@ -22,6 +24,7 @@ import net.minecraft.world.entity.Mob;
+ import net.minecraft.world.entity.MobSpawnType;
+ import net.minecraft.world.entity.SpawnGroupData;
+ import net.minecraft.world.entity.SpawnPlacements;
++import net.minecraft.world.entity.EntitySelector;
+ import net.minecraft.world.level.gameevent.GameEvent;
+ import net.minecraft.world.phys.AABB;
+ import org.slf4j.Logger;
+@@ -46,6 +49,7 @@ public abstract class BaseSpawner {
+     public int requiredPlayerRange = 16;
+     public int spawnRange = 4;
+     private int tickDelay = 0; // Paper
++    private int searchCooldownTicks = 0; // Paper
+ 
+     public BaseSpawner() {}
+ 
+@@ -86,7 +90,24 @@ public abstract class BaseSpawner {
+         tickDelay = world.paperConfig().tickRates.mobSpawner;
+         if (tickDelay == -1) { return; } // If disabled
+         // Paper end
+-        if (this.isNearPlayer(world, pos)) {
++
++        // Paper start - PreSpawnerSearchEvent
++        if (--searchCooldownTicks > 0) return;
++
++        Collection<org.bukkit.entity.Player> activatingPlayers = (Collection)world.getNearbyPlayers(null,
```

Comment:
> I think it might be nicer to have a Collection instead of a List, you can also annotate this constructor with a `@APIStatus.Internal`

Was originally a list but followed the feedback of above. Probably either did it wrong or don't have the write methods in place for a Collection.

Reply:
The problem is not really the collection but more the type ServerPlayer is internal so you can't cast directly that and if you do the right cast (Collection<org.bukkit.entity.Player> with the generic type) things won't compile

Reply:
This is not resolved, please don't close a conversation that is not yet done, this is already the second time, i will not unresolve stuff each day. I'm fine if you need some help however

Reply:
It is, please check the most recent one.
` List<org.bukkit.entity.Player> activatingPlayers = (List)world.getNearbyPlayers(null, (double) pos.getX() + 0.5D, (double) pos.getY() + 0.5D, (double) pos.getZ() + 0.5D, (double) this.requiredPlayerRange, EntitySelector.PLAYER_AFFECTS_SPAWNING); // Uses logic from hasNearbyAlivePlayerThatAffectsSpawning`
This is what it was originally and I have reverted it back to this

Reply:
This is not fixed the way you convert a ServerPlayer to a Bukkit player is using #getBukkitEntity on the internal object. No cast should be needed